ext/geometry/01_geometry.js

```diff
@@ -0,0 +1,2082 @@
+    return matrix;
+  }
+
+  rotate(rotX = 0, rotY, rotZ) {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    rotX = webidl.converters["unrestricted double"](rotX);
+    if (rotY === undefined && rotZ === undefined) {
+      rotZ = rotX;
+      rotX = 0;
+      rotY = 0;
+    } else {
+      rotY = rotY !== undefined
+        ? webidl.converters["unrestricted double"](rotY)
+        : 0;
+      rotZ = rotZ !== undefined
+        ? webidl.converters["unrestricted double"](rotZ)
+        : 0;
+    }
+    const matrix = webidl.createBranded(DOMMatrix);
+    matrix[_writable] = true;
+    matrix[_raw] = new Float64Array(this[_raw]);
+    op_geometry_rotate_self(
+      rotX,
+      rotY,
+      rotZ,
+      matrix[_raw],
+    );
+    matrix[_is2D] = this[_is2D] && rotX === 0 && rotY === 0;
+    return matrix;
+  }
+
+  rotateFromVector(x = 0, y = 0) {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    const matrix = webidl.createBranded(DOMMatrix);
+    matrix[_writable] = true;
+    matrix[_raw] = new Float64Array(this[_raw]);
+    op_geometry_rotate_from_vector_self(
+      webidl.converters["unrestricted double"](x),
+      webidl.converters["unrestricted double"](y),
+      matrix[_raw],
+    );
+    matrix[_is2D] = this[_is2D];
+    return matrix;
+  }
+
+  rotateAxisAngle(x = 0, y = 0, z = 0, angle = 0) {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    x = webidl.converters["unrestricted double"](x);
+    y = webidl.converters["unrestricted double"](y);
+    z = webidl.converters["unrestricted double"](z);
+    const matrix = webidl.createBranded(DOMMatrix);
+    matrix[_writable] = true;
+    matrix[_raw] = new Float64Array(this[_raw]);
+    if (x !== 0 || y !== 0 || z !== 0) {
+      op_geometry_rotate_axis_angle_self(
+        x,
+        y,
+        z,
+        webidl.converters["unrestricted double"](angle),
+        matrix[_raw],
+      );
+    }
+    matrix[_is2D] = this[_is2D] && x === 0 && y === 0;
+    return matrix;
+  }
+
+  skewX(sx = 0) {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    const matrix = webidl.createBranded(DOMMatrix);
+    matrix[_writable] = true;
+    matrix[_raw] = new Float64Array(this[_raw]);
+    op_geometry_skew_self(
+      webidl.converters["unrestricted double"](sx),
+      0,
+      matrix[_raw],
+    );
+    matrix[_is2D] = this[_is2D];
+    return matrix;
+  }
+
+  skewY(sy = 0) {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    const matrix = webidl.createBranded(DOMMatrix);
+    matrix[_writable] = true;
+    matrix[_raw] = new Float64Array(this[_raw]);
+    op_geometry_skew_self(
+      0,
+      webidl.converters["unrestricted double"](sy),
+      matrix[_raw],
+    );
+    matrix[_is2D] = this[_is2D];
+    return matrix;
+  }
+
+  multiply(other = {}) {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    const prefix = "Failed to call 'DOMMatrixReadOnly.prototype.multiply'";
```

Comment:
these prefixes are wrong. they should be "Failed to call '[method]' on '[class]'". the style you used here is only for top level functions or static methods

Reply:
"execute" was used rather than "call", so I changed it to use "execute". In addition: #22579